lisa/sut_orchestrator/libvirt/schema.py

```diff
@@ -52,6 +52,26 @@ class BaseLibvirtPlatformSchema:
     capture_libvirt_debug_logs: bool = False
 
 
+# Configuration options for device-passthrough for the VM.
+@dataclass_json()
+@dataclass
+class LibvirtHostPciDeviceSchema:
```

Comment:
Thanks for addressing all the comments, Smit.

I have another comment in general about the approach.

As I understand it, `device_passthrough` is under node requirements. While this approach works with deploying of a single node, this fails with cases where multiple nodes with passthrough is needed. For example, all network tests will create multiple nodes. And by the current code logic, the deployment of second will fail since all pci devices provided in runbook would be assigned to first node itself.

PCI devices are associated with the host node. And in our context, host node is synonymous to platform. So, we should associate the host pci devices to platform than to node requirement.

Here is what I'm thinking in the form of a sample schema. Introduce 'device_pools' at platform level and define all the host devices there. Pools can be of different types like 'pci_net', 'pci_gpu' etc. All the devices that we want to be used in provisioning guests need to be defined here. And for a node/VM to use a passthrough device, the pool type and devices count has to be mentioned in the node requirement. Platform needs to handle assigning the devices accordingly from the available devices of the pool. Platform needs to maintain the state of the devices (which vm a device is assigned to or not, etc).

```
platform:
  - type: cloud-hypervisor
    cloud-hypervisor:
      network_boot_timeout: $(boot_timeout)
      hosts:
        - address: $(host_address)
          username: $(host_username)
          private_key_file: $(admin_private_key_file)
      capture_libvirt_debug_logs: true
      device_pools:
        - type: "pci_net"
           devices:
             - domain: "1"
                bus: "1"
                slot: "1"
                function: "1"
             - domain: "2"
                bus: "2"
                slot: "2"
                function: "2"
        - type: "pci_gpu"
           devices:
             - domain: "3"
                bus: "3"
                slot: "3"
                function: "3"
             - domain: "4"
                bus: "4"
                slot: "4"
                function: "4"
    requirement:
      core_count: $(guest_vcpu)
      memory_mb: $(guest_memory)
      cloud-hypervisor:
        disk_img: $(qcow2)
        disk_img_format: "qcow2"
        cloud_init:
          extra_user_data: $(extra_user_data)
        device_passthrough:
          - host_device_pool: "pci_net" 
            managed: "yes"
            count: 2
```
I haven't put too much thought into schema. What I listed above is the rough idea. I've also discussed this with @anirudhrb . We can brainstorm more on this offline and finalize the schema.